source/jobpy.py

```diff
@@ -145,39 +156,73 @@ def scrape_indeed_to_pickle(self):
             job = dict([(k,'') for k in MASTERLIST_HEADER])
 
             # scrape the post data
-            job['status'] = 'new'
-            job['title'] = s.find('a', attrs={'data-tn-element': "jobTitle"}).text.strip()
-            job['company'] = s.find('span', attrs={"class":"company"}).text.strip()
-            job['blurb'] = s.find('span',{'class': 'summary'}).text.strip()
-            job['location'] = s.find('span',{'class': 'location'}).text.strip()
-            s.find('span', attrs={'result-link-bar' : 'date'})
-            job['id'] = re.findall(r'id=\"sj_(.*)\" onclick',
-                str(s.find_all('a', { "class" : "sl resultLink save-job-link "})))[0]
-            job['link'] = 'http://www.indeed.{0}/viewjob?jk={1}'.format(
-                           self.search_terms['region']['domain'], job['id'])
+            job['status']   = 'new'
+            try:
+                # jobs should at minimum have a title, company and location
+                job['title']    = s.find('a', attrs={'data-tn-element':'jobTitle'}).text.strip()
+                job['company']  = s.find('span', attrs={'class':'company'}).text.strip()
+                job['location'] = s.find('span', attrs={'class':'location'}).text.strip()
+            except AttributeError:
+                continue
+
+            try:
+                job['blurb']    = s.find('div', attrs={'class':'summary'}).text.strip()
+                # filter all of the weird characters some job postings have...
+                printable = set(string.printable)
+                job['blurb']    = filter(lambda x: x in printable, job['blurb'])
+                job['blurb']    = ''.join(job['blurb'])
+            except AttributeError:
+                job['blurb']    = ""
+
+            try:
+                job['date']     = s.find('span', attrs={'class':'date'}).text.strip()
+            except AttributeError:
+                job['date']     = ""
+
+            try:
+                # sometimes the sl_resultLink_save-job-link class includes a space
+                job['id']       = re.findall(r'id=\"sj_[a-zA-Z0-9]*\"', str(s.find_all('a', attrs={'class':'sl resultLink save-job-link '})[0]))[0]
```

Comment:
This is excellent!!

Can you wrap the code at 80 characters width? This will keep things nice and readable + will match the rest of the codebase + is PEP8.

Reply:
Sure thing.